landoapi/api/try_push.py

```diff
@@ -40,6 +40,18 @@ class PatchFormat(enum.Enum):
     GitFormatPatch = "git-format-patch"
     HgExport = "hgexport"
 
+    def patch_helper(self, patch_io: io.BytesIO) -> PatchHelper:
+        """Return the `PatchHelper` subclass associated with this patch format."""
+        if self == PatchFormat.GitFormatPatch:
+            return GitPatchHelper(patch_io)
+
+        if self == PatchFormat.HgExport:
+            return HgPatchHelper(patch_io)
+
+        raise ValueError(
+            f"Patch format {self.value} has not been mapped to a patch helper."
+        )
```

Comment:
nit: I think it would be cleaner if this was moved outside the enum class. Maybe also rename it to `get_patch_helper`.

Reply:
I renamed the function but I think this makes more sense as an instance method on `PatchFormat`. The function is just a mapping of the patch format to the associated helper class. Writing this as a standalone function turns the usage from `patch_format.get_patch_helper(patch)` to `get_patch_helper(patch_format, patch)`, and moves the mapping outside the enum where it is more relevant.